chore(docs): Update README.md (#18733)
gatsbyjs/gatsby: themes/README.md

> <p align="center">

Reviewer: Please remove the whole header as this is not a public facing README and it creates a bit of churn.

The old header works fine.

```
# Gatsby Themes and Starters

Here are the first party themes maintained by Gatsby.
```

Author: I have done the required changes in the recent commit

  <a href="https://gatsbyjs.org">
    <img alt="Gatsby" src="https://www.gatsbyjs.org/monogram.svg" width="60" />
  </a>
</p>
<h1 align="center">
   Gatsby Themes and Starters
</h1>
<h3 align="center">
  ⚛️ 📄 :rocket:
</h3>
<p align="center">
  <strong>Blazing fast modern site generator for React</strong><br>
  Go beyond static sites: build blogs, e-commerce sites, full-blown apps, and more with Gatsby.
</p>
<p align="center">
  <a href="https://github.com/gatsbyjs/gatsby/blob/master/LICENSE">
    <img src="https://img.shields.io/badge/license-MIT-blue.svg" alt="Gatsby is released under the MIT license." />
  </a>
  <a href="https://circleci.com/gh/DSchau/starters">
    <img src="https://circleci.com/gh/DSchau/starters.svg?style=shield" alt="Current CircleCI build status." />
  </a>
  <a href="https://www.npmjs.org/package/gatsby">
    <img src="https://img.shields.io/npm/v/gatsby.svg" alt="Current npm package version." />
  </a>
  <a href="https://npmcharts.com/compare/gatsby?minimal=true">
    <img src="https://img.shields.io/npm/dm/gatsby.svg" alt="Downloads per month on npm." />
  </a>

>   <a href="https://gatsbyjs.org/docs/how-to-submit-a-pr/">

Reviewer 2: this link is broken ...

Reviewer 2: maybe this link is better then just a deep link to pr?

https://www.gatsbyjs.org/contributing/

Reviewer 2: on homepage its:
```suggestion
  <a href="https://www.gatsbyjs.org/contributing/how-to-contribute/">
```

same broken link here:https://github.com/gatsbyjs/gatsby/blob/master/starters/README.md

    <img src="https://img.shields.io/badge/PRs-welcome-brightgreen.svg" alt="PRs welcome!" />
  </a>
</p>

> ## What's a theme?

Reviewer: Rather than duplicating text and possibly having things out of sync we should link the correct doc page and say that people can learn more about the difference there: https://www.gatsbyjs.org/docs/themes/what-are-gatsby-themes/

- `gatsby-starter-blog-theme` is a starter that uses the blog theme and comes with demo content, etc. [Starters](https://www.gatsbyjs.org/starters/?v=2) are a fairly common concept in the Gatsby ecosystem.
- `gatsby-theme-blog` is a theme that is released as an npm package. You can see this since the `package.json` name is `gatsby-theme-blog` as well.

> See the ["themes"](https://www.gatsbyjs.org/docs/themes/what-are-gatsby-themes/) documenation on the website for more details

Reviewer: ```suggestion
Visit ["What are Gatsby Themes"](https://www.gatsbyjs.org/docs/themes/what-are-gatsby-themes/) for more details.
```

  <a href="https://npmcharts.com/compare/gatsby?minimal=true">
    <img src="https://img.shields.io/npm/dm/gatsby.svg" alt="Downloads per month on npm." />
  </a>

>   <a href="https://www.gatsbyjs.org/contributing/how-to-open-a-pull-request/">

Reviewer 2: the `how to contribute` is the same as in:

https://github.com/gatsbyjs/gatsby/blob/master/README.md
